### DISCUSSION TITLE
What is Smplayer written in?

### DISCUSSION THREAD

**Discussion by mivison** [smplayer-dev/smplayer#48] [Q&A] [answered]
What is Smplayer written in? I would like see how it is done.

**Comment by smplayer-dev** [comment, marked as answer]
It's written in C++. You can see the sources [here](https://github.com/smplayer-dev/smplayer).

**Reply by mivison** [reply]
I have written many scripts for Arduino stuff. I'm fascinated by how you put a frontend on Mplayer. I know this might be a dumb question, but where could a guy get started being able to do that? In my opinion Smplayer blows away VLC by a long shot!! I use all the bells and whistles you put on it to learn new guitar songs, Stuff like the loop option, the speed option, and more. I just maybe needed to be pointed in the right direction.

**Reply by raszpl** [reply]
https://doc.qt.io/qt-5/qtwidgets-tutorials-notepad-example.html
